mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java

```diff
@@ -156,6 +160,34 @@ public boolean onOptionsItemSelected(final MenuItem item) {
                         Toast.makeText(this, getString(R.string.file_browser_is_required), Toast.LENGTH_LONG).show();
                     }
                 }
+            case R.id.menu_plugins_add_url:
+                if (mFileManager.checkWriteStoragePermissionGranted()) {
+                    final AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                    builder.setTitle(R.string.menu_plugins_add_url);
+
+                    final EditText input = new EditText(this);
+                    input.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_URI);
+                    builder.setView(input);
+
+                    builder.setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() { 
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            final String url = input.getText().toString();
+                            final Uri uri = Uri.parse(url);
+                            if (uri != null) {
```

Comment:
The app crashes if I enter an invalid URL (random sets of letters). Seems like this check should be done differently